go/home/home.go

```diff
@@ -523,3 +523,49 @@ func (h *Home) pollOnce(m libkb.MetaContext) (d time.Duration, err error) {
 	}
 	return time.Duration(raw.NextPollSecs) * time.Second, nil
 }
+
+func findBadUserInUsers(m libkb.MetaContext, l []keybase1.HomeUserSummary, badUIDs map[keybase1.UID]bool) bool {
+	for _, s := range l {
+		if badUIDs[s.Uid] {
+			m.Debug("found blocked uid=%s", s.Uid)
+			return true
+		}
+	}
+	return false
+}
+
+func (h *Home) UserBlocked(m libkb.MetaContext, badUIDs map[keybase1.UID]bool) (err error) {
+	h.Lock()
+	defer h.Unlock()
+
+	if !h.peopleCache.hasBadUser(m, badUIDs) {
+		m.Debug("UserBlocked didn't result in any home user suggestions getting blocked, so no-op")
+		return nil
+	}
+
+	h.peopleCache = nil
+	m.Debug("UserBlocked forced home change, updating UI")
+	tmp := h.updateUI(m.Ctx())
+	if tmp != nil {
+		m.Debug("error updating home UI, but ignoring: %s", tmp)
+	}
+	return nil
+}
+
+func (p *peopleCache) hasBadUser(m libkb.MetaContext, badUIDs map[keybase1.UID]bool) bool {
+	if p == nil {
+		m.Debug("nothing to do, people cache is empty")
+		return false
+	}
+
+	if findBadUserInUsers(m, p.all, badUIDs) {
+		m.Debug("Found blocked user in people cache (all)")
+		return true
+	}
+
+	if findBadUserInUsers(m, p.lastShown, badUIDs) {
```

Comment:
Won't `p.lastShown` be covered by `p.all`?

Reply:
I think so, but I wasn't sure.  Figure it didn't hurt to take the conservative approach.

Reply:
Understood, please include a comment mentioning that, since others perusing this code might be confused like I was.